internal/monitors/docker/conversion.go

```diff
@@ -23,25 +21,25 @@ var memoryStatCounters = map[string]bool{
 	"total_pgpgout":    true,
 }
 
-func convertStatsToMetrics(container *dtypes.ContainerJSON, cstats *dtypes.ContainerStats) ([]*datapoint.Datapoint, error) {
-	var parsed dtypes.StatsJSON
-	err := json.NewDecoder(cstats.Body).Decode(&parsed)
-	if err != nil {
-		// EOF means that there aren't any stats, perhaps because the container
-		// is gone.  Just return nothing and no error.
-		if err == io.EOF {
-			return nil, nil
-		}
-		return nil, err
-	}
-	cstats.Body.Close()
-
+func ConvertStatsToMetrics(container *dtypes.ContainerJSON, parsed *dtypes.StatsJSON, enhancedMetrics bool) ([]*datapoint.Datapoint, error) {
```

Comment:
Pass in that embedded config struct with all the flags and the split it out here to the various `convert*` functions.